src/java.base/share/classes/java/lang/ref/package-info.java

```diff
@@ -93,6 +93,28 @@
  * structure, this check will add little overhead to the hashtable
  * access methods.
  *
+ * <h3>Memory Consistency Properties</h3>
+ * Certain interactions between the garbage collector, references, and reference
+ * queues form
+ * <a href="{@docRoot}/java.base/java/util/concurrent/package-summary.html#MemoryVisibility"><i>happens-before</i></a>
+ * relationships:
+ *
+ * <ul>
+ *
+ * <li> Actions in a thread prior to calling
```

Comment:
Is line 104 needed? Can we just say that Refrence.reachaboilityFence(x) happens-before ...